### ISSUE TITLE
lots of "unhandled" messages when using svgs from Sketch

### ISSUE THREAD

**Issue by SteveAlexander** [dnfield/flutter_svg#58]
I'm using a few SVGs exported from Sketch.

I get lots of debug messages about three unhandled elements:

- `<title>`
- `<mask>`
- `<polygon>`

example:
```
flutter: unhandled element polygon; Picture key: AssetBundlePictureKey(bundle: PlatformAssetBundle#4c5d9(), name: "assets/microphone.svg", colorFilter: ColorFilter(Color(0xffffffff), BlendMode.srcIn))
```

While the SVGs work fine in Flutter with flutter_svg for my purposes, and I can edit the SVGs and remove or comment out these elements, this does have an impact of the signal-to-noise ratio when working on the app if I don't edit the SVGs.

I see there is a print statement on line 96 of `svg_parser.dart`:

```
  } else if (el.name.local != 'desc') {
    // no plans to handle these
    print('unhandled element ${el.name.local}; Picture key: $key');
  }
```

I have some ideas what to do about this, and I'll be happy to provide a PR for one of the following:
1. remove the print statement altogether
2. record a set of (el.name.local, key) and print each unique error only once per runtime
3. specifically ignore title, polygon, mask, so that regular SVG exports from Sketch aren't noisy.

Thanks

**Comment by dnfield** [comment]
That sounds good to me.

I think polygon should be supported tho, that seems like a bug.

**Comment by SteveAlexander** [comment]
I guess `<title>` might just be a no-op, like `<desc>` appears to be?

**Comment by SteveAlexander** [comment]
I think the polygon issue is when it is in a `<defs>` section: 

```
    <defs>
        <polygon id="path-1" points="0 14 7.46666667 14 7.46666667 0 0 0"></polygon>
    </defs>
```

This gets referred to later:

```
                        <mask id="mask-2" fill="white">
                            <use xlink:href="#path-1"></use>
                        </mask>

```

**Comment by dnfield** [comment]
I've had some thoughts o ftrying to use `<title>` (but desc doesn't seem as useful) for a11y stuff, but nothing too concrete.

There's also been at least some discussion around supporting shapes in `<defs>`, but nothing concrete there yet either.

I would want to avoid adding too much overhead just to limit verbosity, but if all we're talking about is a simple hashset of elements I think that's probably ok. Of course, it'd be possible to preprocess SVGs to remove these kinds of elements too with XSLT or somesuch, which would add no overhead to the parser for it.

**Comment by SteveAlexander** [comment]
I guess an "avoid verbose repeated output" switch could be on only in debug mode — a bit like the error reporting in flutter core that abbreviates repeated errors.

**Comment by dnfield** [comment]
I've checked in a fix for this in 000e17f - wrapped in an assert, so no impact on release code.

It will now track whether it's printed about a particular unhandled element and only print on the first time.

**Comment by SteveAlexander** [comment]
awesome — I didn't think to use `assert` in this way. Thanks!
